ArkhamOverlay/Pages/SelectCards/SelectCardsController.cs

```diff
@@ -109,31 +157,53 @@ public class SelectCardsController : Controller<SelectCardsView, SelectCardsView
             var commands = new List<RightClickOptionCommand>();
             foreach (var option in options) {
-                var text = option.GetTextResolvingPlaceholders(ResolveMenuItemPlaceholder);
+                var text = option.GetText(this);
                 if (!string.IsNullOrEmpty(text)) {
-                    commands.Add(new RightClickOptionCommand(option.Option, text, callback));
+                    commands.Add(new RightClickOptionCommand(option, text, callback));
                 }
             }
 
             return commands;
         }
 
         /// <summary>
-        /// Callback to resolve placeholder in a menu item so we can provide more contextual information
+        /// Used by button option to get a name for the card group when displaying an option
+        /// </summary>
+        /// <param name="cardGroupId">Card group to resolve</param>
+        /// <returns>The name of the card group</returns>
+        string IButtonOptionResolver.GetCardGroupName(CardGroupId cardGroupId) {
+            var cardGroup = _appData.Game.GetCardGroup(cardGroupId);
+            return cardGroup.Name;
+        }
+
+        /// <summary>
+        /// Used by button option to get a name for the card zone when displaying an option
         /// </summary>
-        /// <param name="placeholder">The placeholder to resolve</param>
-        /// <returns>The actual value the placeholder represents</returns>
-        /// <remarks>Example "player1" (represented by <<xxxx>></xxxx> in the text) will resolve to the name of player 1 in game data</remarks>
-        private string ResolveMenuItemPlaceholder(string placeholder) {
-            if (Enum.TryParse(placeholder, out CardGroupId cardGroupId)) {
-                var cardGroup = _appData.Game.GetCardGroup(cardGroupId);
-                return cardGroup.Name;
+        /// <param name="cardGroupId">Card group of the card zone</param>
+        /// <param name="zoneIndex">Zone to resolve</param>
+        /// <returns>Name of the zone</returns>
+        string IButtonOptionResolver.GetCardZoneName(CardGroupId cardGroupId, int zoneIndex) {
+            var cardGroup = _appData.Game.GetCardGroup(cardGroupId);
+            var cardZone = cardGroup.GetCardZone(zoneIndex);
+            if (cardZone == default) {
+                return string.Empty;
             }
+            return cardZone.Name;
+        }
 
-            return null;
+        /// <summary>
+        /// Used by button option to get the image ID for a button when displaying an option
+        /// </summary>
+        /// <param name="cardGroupId">Card group of the card zone</param>
+        /// <param name="zoneIndex">Zone to resolve</param>
+        /// <returns>Image Id for the card group</returns>
+        string IButtonOptionResolver.GetImageId(CardGroupId cardGroupId, int zoneIndex) {
+            var cardGroup = _appData.Game.GetCardGroup(cardGroupId);
+            return cardGroup.Name;
```

Comment:
Is Name the correct field to return here?

Reply:
It is for the moment- we are using the name of the investigator for the "image id" of the card group. This may change in the future.